test/saithrift/tests/sail3.py

```diff
@@ -807,6 +807,244 @@ def runTest(self):
 
             self.client.sai_thrift_remove_virtual_router(vr1)
 
+            
+@group('l3')
+@group('ecmp')
+class L3IPv4EcmpHashSeedTest(sai_base_test.ThriftInterfaceDataPlane):
+    def runTest(self):
+        '''
+	Create a VRF with IPv4 and IPv6 enabled. Create 4 router interfaces in the same VRF.
+        Create a route (/24 mask) with nhop and neighbor entry in three router interfaces. 
+        Send 100 streams with varying 5-tuple combinations to the destination IP on one port and verify distribution on the 
+        three router interfaces for which the nhops are present. Change the ECMP hash seed value to 10 and verify distribution.
+	   
+	'''
+	   
+        print "Sending packet port4 -> port1,port2,port3 (192.168.6.1 to 10.10.10.1) "
+        switch_init(self.client)
+        port1 = port_list[0]
+        port2 = port_list[1]
+        port3 = port_list[2]
+	port4 = port_list[3]
+        v4_enabled = 1
+        v6_enabled = 1
+        mac = ''
+        hash_seed = 10
+
+        addr_family = SAI_IP_ADDR_FAMILY_IPV4
+        
+        ip_addr1 = '192.168.1.1'
+        ip_addr2 = '192.168.2.1'
+        ip_addr3 = '192.168.3.1'
+        ip_addr4 = '10.10.10.0'
+        
+        ip_addr1_subnet = '192.168.1.0'
+        ip_addr2_subnet = '192.168.2.0'
+        ip_addr3_subnet = '192.168.3.0'
+       
+        ip_mask = '255.255.255.0'
+        destmac1 = '00:11:22:33:44:51'
+        destmac2 = '00:11:22:33:44:52'
+	destmac3 = '00:11:22:33:44:53'
+
+        virtual_router = sai_thrift_create_virtual_router(self.client, v4_enabled, v6_enabled)
+
+        router_interface1 = sai_thrift_create_router_interface(self.client, virtual_router, SAI_ROUTER_INTERFACE_TYPE_PORT, port1, 0, v4_enabled, v6_enabled, mac)
+        router_interface2 = sai_thrift_create_router_interface(self.client, virtual_router, SAI_ROUTER_INTERFACE_TYPE_PORT, port2, 0, v4_enabled, v6_enabled, mac)
+        router_interface3 = sai_thrift_create_router_interface(self.client, virtual_router, SAI_ROUTER_INTERFACE_TYPE_PORT, port3, 0, v4_enabled, v6_enabled, mac)
+	router_interface4 = sai_thrift_create_router_interface(self.client, virtual_router, SAI_ROUTER_INTERFACE_TYPE_PORT, port4, 0, v4_enabled, v6_enabled, mac)
+
+        sai_thrift_create_neighbor(self.client, addr_family,  router_interface1, ip_addr1, destmac1)
+        sai_thrift_create_neighbor(self.client, addr_family,  router_interface2, ip_addr2, destmac2)
+        sai_thrift_create_neighbor(self.client, addr_family,  router_interface3, ip_addr3, destmac3)
+
+        next_hop1 = sai_thrift_create_nhop(self.client, addr_family, ip_addr1, router_interface1)
+        next_hop2 = sai_thrift_create_nhop(self.client, addr_family, ip_addr2, router_interface2)
+	next_hop3 = sai_thrift_create_nhop(self.client, addr_family, ip_addr3, router-interface3)
```

Comment:
'router-interface3' should be 'router_interface3'

Reply:
changes are made as per the comments